### PR TITLE
Fix Issue 5897 - Docker python scripts use Context if available

### PR DESCRIPTION
**Pull request by beldcode**
include context file (zap docker scan)

### CODE REVIEW

**Comment by kingthorin** [comment]
Please start with a properly up to date base branch.

**Comment by thc202** [comment]
Now ready for reviews :)

**Comment by kingthorin** [comment]
Thank you both!!!

File: docker/zap_common.py [hunk 1]
```diff
@@ -382,23 +387,32 @@ def zap_spider(zap, target):
...
+'''
+In zap_active_scan, we use the global context_id, which is used in zap.ascan.scan to indicate that there is
+a context
+'''
 @hook(wrap=True)
 def zap_active_scan(zap, target, policy):
     logging.debug('Active Scan ' + target + ' with policy ' + policy)
-    ascan_scan_id = zap.ascan.scan(target, recurse=True, scanpolicyname=policy)
+    if context_id is None:
+        ascan_scan_id = zap.ascan.scan(target, recurse=True, scanpolicyname=policy)#, contextid=None)
```
Comment (psiinon) [inline docker/zap_common.py:413]: This looks a bit weird. Cant we just use contextid=context_id and drop the test?

File: docker/zap_common.py [hunk 2]
```diff
@@ -382,23 +387,32 @@ def zap_spider(zap, target):
 
 
 @hook(wrap=True)
-def zap_ajax_spider(zap, target, max_time):
+def zap_ajax_spider(zap, target, max_time, context=None):
     logging.debug('AjaxSpider ' + target)
     if max_time:
         zap.ajaxSpider.set_option_max_duration(str(max_time))
-    zap.ajaxSpider.scan(target)
+    if context is not None:
```
Comment (psiinon) [inline docker/zap_common.py:394]: Is this test needed or can we pass across context=None?

File: docker/zap-baseline.py [hunk 3]
```diff
@@ -321,10 +321,23 @@ def main(argv):
         time.sleep(2)
 
         # Spider target
-        zap_spider(zap, target)
+        '''
+        Here the zap spider is supposed to only use a context if a context file is given
```
Comment (thc202) [inline docker/zap-baseline.py:325]: I'd suggest removing this comment, it's repeating what the code is doing (same for the others).

File: docker/zap_common.py [hunk 4]
```diff
@@ -510,8 +524,14 @@ def write_report(file_path, report):
...
         f.write(report)
 
+'''
+zap_import_context imports the context and stores a context_id which will be used by zap_active_scan. Therefore the
+context_id must be global so that zap_active_scan has access to the variable.
+'''
+
 @hook(wrap=True)
 def zap_import_context(zap, context_file):
+    global context_id
```
Comment (thc202) [inline docker/zap_common.py:534]: The ID is being returned so the caller can pass the ID when calling the other function.

File: docker/zap-baseline.py [hunk 5]
```diff
@@ -321,10 +321,23 @@ def main(argv):
...
+        otherwise the inital spider funcionality without a context is used
+        '''
+        if context_file:
+            zap_spider(zap, target, os.path.splitext(os.path.basename(context_file))[0])
```
Comment (thc202) [inline docker/zap-baseline.py:329]: The name of the context might not be the same as the name of the file, better get the name from ZAP (e.g. last one in the context list), it also avoids processing the name multiple times.

Comment (kingthorin) [reply docker/zap-baseline.py:329]: Good catch.

**Comment by thc202** [comment]
The additional parameter in `zap_spider` and `zap_ajax_spider` will break the hooks when using a context, e.g.:
```
TypeError: zap_spider() takes 2 positional arguments but 3 were given
```

I guess we'll need to use the global variable in the meantime to get the context name in the spider functions.

**Comment by kingthorin** [comment]
9c5d70d seems to have included commented code instead of removing it, and a bunch of un-related files got line ending changes? Also the docker file name probably should change.

File: docker/zap_common.py (hunk 1 above)
Comment (beldcode) [reply docker/zap_common.py]: I think we could use **kwargs since the argument "contextid" can appear or not. This way the test could be dropped..

Comment (beldcode) [reply docker/zap_common.py]: We could use **kwargs in the zap_active_scan function?

Comment (beldcode) [reply docker/zap_common.py]: the context_id variable is initialized in the zap_import_context function.. this function is only called if a context_file is given. So context_id is not always available..

**Comment by kingthorin** [comment]
> Also the docker file name probably should change.

I made a typo that should have said **"shouldn't"**!

File: docker/zap-baseline.py (hunk 5 above)
Comment (beldcode) [reply docker/zap-baseline.py:329]: I'll use ZAP for that! Thx for the hint.

**Comment by beldcode** [comment]
> al

File: docker/zap-baseline.py (hunk 5 above)
Comment (beldcode) [reply docker/zap-baseline.py:329]: @thc202 is there a special reason why to use the last context name from the context list?

Comment (thc202) [reply docker/zap-baseline.py:329]: The last one is the one that was imported/added by the script.

File: docker/zap_common.py (hunk 1 above)
Comment (thc202) [reply docker/zap_common.py]: It's fine if it's `None`, the API client will just ignore that parameter.

**Comment by thc202** [comment]
Rebased to fix the conflicts and bring the latest changes. No functional changes done yet.

**Comment by thc202** [comment]
Addressed the review comments.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
@beldcode thanks!

**Pull request merged**